internal/impl/devbox.go

```diff
@@ -232,16 +232,18 @@ func (d *Devbox) Shell() error {
 		return err
 	}
 
-	env, err := plugin.Env(d.cfg.Packages, d.projectDir)
-	if err != nil {
-		return err
-	}
-
+	var env []string
 	if featureflag.NixlessShell.Enabled() {
+		// computeNixEnv() includes plugin env setup
 		env, err = d.computeNixEnv()
 		if err != nil {
 			return err
 		}
+	} else {
+		env, err = plugin.Env(d.cfg.Packages, d.projectDir)
```

Comment:
If you replace this with `env = append(envPairs, d.getConfigEnvs(envPairs)...)` then you are not blocked by nixsless shell flag right?

Reply:
oh that's right! I'll make the change

Reply:
I ended up removing this else because it deviates from the logic we have in devbox run. I think it's ok to be blocked by unified_env feature flag